drivers/ma_yya_report/app/models/ma_yya_report/report.rb

```diff
@@ -0,0 +1,229 @@
+        D2b: a_t[:race].eq(3),
+        D2c: a_t[:race].eq(2),
+        D2d: a_t[:race].eq(1),
+        D2e: a_t[:race].eq(4),
+        D2f: a_t[:race].in([6, 8, 9, 99]),
+        D2g: a_t[:ethnicity].eq(1),
+        D2h: nil,
+        D2i: nil,
+        D2j: nil,
+
+        D3a: a_t[:mental_health_disorder].eq(true),
+        D3b: a_t[:substance_use_disorder].eq(true),
+        D3c: a_t[:physical_disability].eq(true),
+        D3d: a_t[:developmental_disability].eq(true),
+
+        D4a: a_t[:pregnant].eq(1).and(a_t[:due_date].gt(report_start_date)).
+          or(a_t[:head_of_household].eq(true).and(Arel.sql(custodial_parent_query))),
+        D4b: a_t[:sexual_orientation].in([2, 3, 4, 5]).or(a_t[:gender].eq(5)),
+        D4c: a_t[:education_status_date].lteq(report_end_date).
+          and(a_t[:current_school_attendance].eq(0)).and(a_t[:most_recent_education_status].in([0, 1])),
+        D4d: a_t[:health_insurance].eq(true),
+
+        Ea: nil,
+        Eb: nil,
+
+        F1a: a_t[:subsequent_current_living_situations].not_eq('[]').and(a_t[:reported_previous_period].eq(false)),
+        F1b: a_t[:reported_previous_period].eq(false).
+          and(Arel.sql(
+                json_contains(:subsequent_current_living_situations,
+                              [15, 6, 7, 25, 4, 5, 29, 14, 32, 36, 35, 28, 19, 3, 31, 33, 34, 10, 20, 21, 11]),
+              )),
+
+        F2a: f2_population.
+          and(a_t[:rehoused_on].between(report_start_date..report_end_date)),
+        F2b: f2_population.
+          and(a_t[:rehoused_on].between(report_start_date..report_end_date)).
+          and(a_t[:subsequent_current_living_situations].not_eq('[]')),
+        F2c: f2_population.
+          and(a_t[:rehoused_on].not_eq(nil)).
+          and(Arel.sql(json_contains(:subsequent_current_living_situations, [19, 3, 31, 33, 34, 10, 20, 21, 11]))),
+        F2d: nil,
+
+        G1a: g_population.and(a_t[:age].lt(18)),
+        G1b: g_population.and(a_t[:gender].eq(1)),
+        G1c: g_population.and(a_t[:gender].eq(0)),
+        G1d: g_population.and(a_t[:gender].eq(5)),
+        G1e: g_population.and(a_t[:gender].in([4, 6, 8, 9, 99])),
+
+        G2a: g_population.and(a_t[:race].eq(5)),
+        G2b: g_population.and(a_t[:race].eq(3)),
+        G2c: g_population.and(a_t[:race].eq(2)),
+        G2d: g_population.and(a_t[:race].eq(1)),
+        G2e: g_population.and(a_t[:race].eq(4)),
+        G2f: g_population.and(a_t[:race].in([6, 8, 9, 99])),
+        G2g: g_population.and(a_t[:ethnicity].eq(1)),
+
+        G3a: g_population.and(a_t[:sexual_orientation].in([2, 3, 4, 5]).or(a_t[:gender].eq(5))),
+      }.freeze
+    end
+
+    private def custodial_parent_query
+      'jsonb_array_length(household_ages) > 1' +
+        'AND EXISTS(SELECT jsonb_array_elements(household_ages) AS age WHERE age < 18)'
+    end
+
+    private def json_contains(field, contents)
+      contents.map { |val| "#{field} @> '#{val}'" }.join(' OR ')
+    end
```

Comment:
Isn't there something in `active_record_extended` that covers this?  Nothing wrong with this, but I wonder if we could lean on that gem.

Reply:
I couldn't get it to work with JSON -- it is documented for Array fields, maybe I can just use PG arrays?

Reply:
This is fine.  Nothing fancy going on, just thought I'd ask.